levels/level02.ts

```diff
@@ -0,0 +1,162 @@
+game.registerLevel({
+  level: 2,
+  newTown: (game: Game) => {
+    const town = new Town();
+    const boss = new Boss();
+
+    town.name = 'Magma Town';
+    town.townsfolk = 200;
+    town.foodStock = 150;
+    town.foodSupport = [3, 4, 3, 1];
+    town.foodCostBuy = [3, 3, 4, 6];
+    town.foodCostSell = [1, 1, 1, 4];
+    town.waterStock = 150;
+    town.waterSupport = [4, 5, 5, 4];
+    town.waterCostBuy = [3, 3, 3, 3];
+    town.waterCostSell = [2, 2, 2, 2];
+    for (const cat of EQ_FINE_CATEGORIES) {
+      town.inventoryWeapon[cat] = 100;
+      town.inventoryWeaponBuy[cat] = 5;
+      town.inventoryWeaponSell[cat] = 3;
+      town.inventoryArmor[cat] = 100;
+      town.inventoryArmorBuy[cat] = 5;
+      town.inventoryArmorSell[cat] = 3;
+    }
+    town.need = 45;
+    town.needMax = 45;
+    town.needRatio = 0.010;
+    town.boss = 3000;
+    town.bossReward = 300;
+
+    // Town State:
+
+    function maybeAngerGods(game: Game) {
+      if (!game.party.status.angeredGods.active) {
+        // Being charasmatic helps you avoid making a faux pas at a local ceremony. 
+        const r = rollDie(20) + calcmod(game.party.cha, [[0, -1], [5, 0], [14, 1]]);
+        if (r <= 2) {
+          game.party.status.angeredGods.active = true;
+          setStatusExpiry(game, game.party.status.angeredGods, { term: 75 });
+          game.log('A party member commits a faux pas at a ceremony with some townsfolk.');
+        } else if (r <= 4) {
+          if (game.party.wis >= 14) {
```

Comment:
```suggestion
          if (game.party.cha >= 14) {
```

Reply:
I commited this change from github, then realized after reviewing I actually intended to test wisdom, not charisma. Changed back in a future commit.